rest_framework/utils/serializer_helpers.py

```diff
@@ -28,6 +29,22 @@ def __reduce__(self):
         # but preserve the raw data.
         return (dict, (dict(self),))
 
+    if sys.version_info >= (3, 9):
```

Comment:
Is there a reason for the version gate here?
The methods could be always defined, even if only python >= 3.9 uses them.

Reply:
It's not a case of only Python 3.9 using them - if you define those methods, then `|` will do a dictionary merging like it does in 3.9. So the question is - if I'm on Python 3.8, what do I expect these would do:

```python
>>> {} | {}
>>> serializer_instance.data | {}
```

On Python 3.8, the first one will produce `TypeError: unsupported operand type(s) for |: 'dict' and 'dict'` and there is nothing we can do about that. So I expect most people would expect the second to do that as well. 

I think we are asking for confusion and trouble if we effectively backport PEP 584 to earlier Python versions, but only for the return value of `Serializer.data`, and nowhere else (because we can't).